src/election_anomaly/user_interface/__init__.py

```diff
@@ -730,16 +767,23 @@ def get_runtime_parameters(keys, param_file=None):
 	if len(p) == 0:
 		raise FileNotFoundError
 
-	for k in keys:
+	for k in required_keys:
 		try:
 			d[k] = parser['election_anomaly'][k]
 		except KeyError:
-			missing_params['missing'].append(k)
+			missing_required_params['missing'].append(k)
+
+	if optional_keys:
+		for k in optional_keys:
+			try:
+				d[k]=parser['election_anomaly'][k]
+			except KeyError:
+				d[k] = None
 
-	if len(missing_params['missing']) == 0:
-		missing_params = None
+	if len(missing_required_params['missing']) == 0:
```

Comment:
Can we test with `if not missing_required_params['missing']:`? Again, somewhat of a comment on the style of coding, but I believe our checking for empty lists in this codebase typically uses truthiness/falsiness as opposed to checking length.

Reply:
Done